modules/apps/portal-security/portal-security-content-security-policy-test/src/testIntegration/java/com/liferay/portal/security/content/security/policy/internal/servlet/filter/test/ContentSecurityPolicyFilterTest.java

```diff
@@ -0,0 +1,257 @@
+				httpClient.getHeaderField(
+					ContentSecurityPolicyConstants.
+						CONTENT_SECURITY_POLICY_HEADER),
+				cspPolicy);
+		}
+
+		_configureContentSecurityPolicy(true, cspPolicy);
+	}
+
+	@Test
+	public void testWhenConfiguredCorrectlyNonceShouldBeGeneratedToBodyAndHeader()
+		throws Exception {
+
+		String cspPolicy =
+			"default-src 'self'; script-src 'self' '[nonceToken]'; style-src " +
+				"'self' '[nonceToken]'";
+
+		try (CompanyConfigurationTemporarySwapper
+				configurationTemporarySwapper = _configureContentSecurityPolicy(
+					true, cspPolicy)) {
+
+			HttpURLConnection httpClient = _executePortalCall();
+
+			Map<String, List<String>> headerFields =
+				httpClient.getHeaderFields();
+
+			Assert.assertTrue(
+				headerFields.containsKey(
+					ContentSecurityPolicyConstants.
+						CONTENT_SECURITY_POLICY_HEADER));
+
+			String contentSecurityPolicyHeaderValue = httpClient.getHeaderField(
+				ContentSecurityPolicyConstants.CONTENT_SECURITY_POLICY_HEADER);
+
+			Assert.assertNotNull(contentSecurityPolicyHeaderValue);
+
+			int nonceStartPos = contentSecurityPolicyHeaderValue.indexOf(
+				ContentSecurityPolicyConstants.
+					CONTENT_SECURITY_POLICY_NONCE_HEADER_PREFIX);
+
+			int noncePrefixLength =
+				ContentSecurityPolicyConstants.
+					CONTENT_SECURITY_POLICY_NONCE_HEADER_PREFIX.length();
+
+			int nonceRandomStartPos = nonceStartPos + noncePrefixLength;
+
+			String nonce = contentSecurityPolicyHeaderValue.substring(
+				nonceRandomStartPos, nonceRandomStartPos + 24);
+
+			String substitutedCspPolicy = StringUtil.replace(
+				cspPolicy,
+				ContentSecurityPolicyConstants.
+					CONTENT_SECURITY_POLICY_NONCE_TOKEN,
+				ContentSecurityPolicyConstants.
+					CONTENT_SECURITY_POLICY_NONCE_HEADER_PREFIX + nonce);
+
+			Assert.assertEquals(
+				contentSecurityPolicyHeaderValue, substitutedCspPolicy);
+
+			String responseBody = _getResponseBody(httpClient);
+
+			Assert.assertTrue(
+				responseBody.contains("<link nonce='" + nonce + "'"));
+
+			Assert.assertTrue(
+				responseBody.contains("<script nonce='" + nonce + "'"));
+
+			Assert.assertTrue(
+				responseBody.contains("<style nonce='" + nonce + "'"));
+		}
+	}
+
+	private CompanyConfigurationTemporarySwapper
+			_configureContentSecurityPolicy(boolean enabled, String policy)
+		throws Exception {
+
+		return new CompanyConfigurationTemporarySwapper(
+			TestPropsValues.getCompanyId(),
+			"com.liferay.portal.security.content.security.policy.internal." +
+				"configuration.ContentSecurityPolicyConfiguration",
+			HashMapDictionaryBuilder.<String, Object>put(
+				"enabled", enabled
+			).put(
+				"policy", policy
+			).build(),
+			SettingsFactoryUtil.getSettingsFactory());
+	}
+
+	private HttpURLConnection _executePortalCall() throws IOException {
+		URL url = new URL("http://localhost:8080/web/guest");
+
+		HttpURLConnection httpClient = (HttpURLConnection)url.openConnection();
+
+		httpClient.setRequestMethod("GET");
+
+		return httpClient;
+	}
```

Comment:
This method actually does not execute the portal call. It is not until you call a method `HttpURLConnection#connect` . It also happens when a method is returning something from the HTTP response, such `HttpURLConnection#getHeaderFields`, that the HTTP request is actually sent.

So the method name should probably be getHttpURLConnection (and the local variable renamed to match also).